feat(ssr): `omitLineBreaks` option

Mini-ghost opened the pull request:
### 🔗 Linked issue

https://github.com/nuxt/nuxt/pull/24888

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I'm working to minimize the size of a Nuxt-rendered HTML archive, and I've observed that tags inside the `<head>` include `\n`, potentially adding a few extra kilobytes for larger sites.

I'm uncertain about the purpose of these line breaks - perhaps for enhanced readability? When I removed the newlines, my project size decreased by 1kb. It's a small reduction, but could this be significant for large-scale users?

Should you deem this PR more detrimental than beneficial, or have other reservations, please feel free to close it. Thank you.

---

After discussion, I have introduced `options.omitLineBreaks` to configure the retention of line breaks. This approach preserves the ability to easily debug page source, while also empowering developers to compress the content within `<head>` into a single line.

**`options.omitLineBreaks` is `false` or `undefined`**

```html
<head><meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="preload" as="fetch" crossorigin="anonymous" href="/_payload.json">
<link rel="modulepreload" as="script" crossorigin href="/_nuxt/entry.lirqHv3X.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/error-404.tbGoJl1q.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/vue.f36acd1f.eXCIzdvm.js">
<link rel="prefetch" as="script" crossorigin href="/_nuxt/error-500.9n_HQChl.js">
<script type="module" src="/_nuxt/entry.lirqHv3X.js" crossorigin></script></head>
```

**`options.omitLineBreaks` is `true`**

```html
<head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="preload" as="fetch" crossorigin="anonymous" href="/_payload.json"><link rel="modulepreload" as="script" crossorigin href="/_nuxt/entry.lirqHv3X.js"><link rel="prefetch" as="script" crossorigin href="/_nuxt/error-404.tbGoJl1q.js"><link rel="prefetch" as="script" crossorigin href="/_nuxt/vue.f36acd1f.eXCIzdvm.js"><link rel="prefetch" as="script" crossorigin href="/_nuxt/error-500.9n_HQChl.js"><script type="module" src="/_nuxt/entry.lirqHv3X.js" crossorigin></script></head>
```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

harlan-zw commented:
Thanks for the PR :) Have replied here: https://github.com/nuxt/nuxt/pull/24888#issuecomment-1869286993

Mini-ghost commented:
@harlan-zw 

Following the discussion in https://github.com/nuxt/nuxt/pull/24888, I have added the `options.omitLineBreaks` option. Could you kindly review it to see if it meets your expectations?

Please feel free to share any suggestions you may have. Thank you.

harlan-zw commented on packages/schema/src/hooks.ts:
```diff
@@ -12,6 +12,10 @@ export interface SSRHeadPayload {
   bodyAttrs: string
 }
 
+export interface SSRHeadOptions {
```
Can you please name this `RenderSSRHeadOptions`

harlan-zw commented:
Great work @Mini-ghost, one small name change in the code.

I need to improve the doc structure, but for the time being would are you be able to add some details on usage for this within this section https://unhead.unjs.io/setup/ssr/installation#_1-generating-the-ssr-payload. Something like

```md
### Options

*omitLineBreaks* 

- Type: `boolean`
- Default: `false`

Render the head tags without line breaks.
```

Mini-ghost commented:
@harlan-zw Thank you for your guidance. I have updated the documentation and added a bit more detail in the description, hopefully it doesn't make the document overly complicated.

harlan-zw commented:
Perfect, thanks again.

Since this is a new feature I'll need to introduce this in a minor release so may try and bundle up a couple of other PRs before this is released.
